airflow/providers/amazon/aws/operators/emr.py

```diff
@@ -191,8 +196,23 @@ def __init__(
         aws_conn_id: str = "aws_default",
         waiter_countdown: int = 25 * 60,
         waiter_check_interval_seconds: int = 60,
+        # TODO: waiter_max_attempts and waiter_delay should default to None when the other two are deprecated.
+        waiter_max_attempts: int | ArgNotSet = NOTSET,
+        waiter_delay: int | ArgNotSet = NOTSET,
         **kwargs: Any,
     ):
+        if waiter_max_attempts is NOTSET:
+            warnings.warn(
+                "The parameter waiter_countdown has been deprecated to standardize "
+                "naming conventions.  Please use waiter_max_attempts instead.  In the "
+                "future this will default to None and defer to the waiter's default value."
+            )
```

Comment:
```suggestion
        if waiter_max_attempts is NOTSET:
            warnings.warn(
                "The parameter waiter_countdown has been deprecated to standardize "
                "naming conventions.  Please use waiter_max_attempts instead.  In the "
                "future this will default to None and defer to the waiter's default value."
            )
            waiter_max_attempts = waiter_countdown // waiter_check_interval_seconds
```

What if we change value here  so we could change this

```python
self.waiter_max_attempts = (
            (waiter_countdown // waiter_check_interval_seconds)
            if waiter_max_attempts is NOTSET
            else waiter_max_attempts
        )
```
to

```python
self.waiter_max_attempts = waiter_max_attempts
```

Reply:
And the same to waiter_delay

Reply:
I'm not sure I follow.    `waiter_countdown` and `waiter_check_interval_seconds` both already exist and we are deprecating those, but we can't just ignore them or that would break everyone currently using them.

Reply:
It is not about ignore deprecated one, just suggestion for calculate `waiter_max_attempts` on own condition so we do not need to check `waiter_max_attempts is NOTSET` twice.

But anyway it is just minor suggestion not affect to anything critical, we could keep it as it is.

Reply:
I totally missed that you were assigning the value inside the deprecation block :facepalm: Yeah, that makes perfect sense and I'll make the change.... after I grab another coffee, apparently...    Sorry about that.